web/html/src/components/steps-progress-bar.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from "react";
+import { useState } from "react";
+
+import { Button } from "./buttons";
+
+type Step = {
+  title: string;
+  content: React.ReactNode;
+  validate: boolean | null;
```

Comment:
This seems curious to me, but maybe I don't understand the idea. Is there a reason not to use `validate?: boolean` or similar? That way if there is no validation, the field can be omitted altogether.